src/Data/Singletons/Single.hs

```diff
@@ -590,7 +590,7 @@ sEqToSDecide p@(DConPr n)
 sEqToSDecide DWildCardPr = DWildCardPr
 
 isException :: DExp -> Bool
-isException (DVarE n)             = n == undefinedName
+isException (DVarE n)             = nameBase n == "sUndefined"
```

Comment:
By "do better", do you mean check if `nameBase n == nameBase 'sUndefined`? I don't think we can check if it's directly equal to ``sUndefined`, since I believe we're checking a `Name` which has been stubbed out by Template Haskell, so it won't have the right namespace.